### PR TITLE
Fixed #34046 -- Fixed pinning flake8 and isort versions in tox.ini.

### PR DESCRIPTION
**Pull request by icemac**
Now they run the tools expected.

Additionally I set the basepython version to the minimum supported Python version to prevent surprises running the tests when python3 points to an older Python version.

Ticket: https://code.djangoproject.com/ticket/34046

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -9,18 +9,23 @@ skipsdist = true
...
     py3
     black
-    flake8 >= 3.7.0
+    flake8
     docs
-    isort >= 5.1.0
+    isort
 
-# Add environment to use the default python3 installation
+# Add environment to use the minmum supported python3 installation
```
Comment (felixxm) [inline tox.ini:16]:
```suggestion
# Add environment to use the minmum supported python3 installation.
```

File: tox.ini [hunk 2]
```diff
@@ -9,18 +9,23 @@ skipsdist = true
...
 [testenv:py3]
-basepython = python3
+basepython = python3.8
```
Comment (felixxm) [inline tox.ini:18]: I'd revert this and the comment change. `Tox` is not used by CI, and there is not need to force users to use e.g. `3.8` instead of `3.10`.

File: tox.ini [hunk 3]
```diff
@@ -9,18 +9,23 @@ skipsdist = true
...
 [testenv]
 usedevelop = true
 # OBJC_DISABLE_INITIALIZE_FORK_SAFETY fixes hung tests for MacOS users. (#30806)
-passenv = DJANGO_SETTINGS_MODULE PYTHONPATH HOME DISPLAY OBJC_DISABLE_INITIALIZE_FORK_SAFETY
+passenv =
+    DISPLAY
+    DJANGO_SETTINGS_MODULE
+    HOME
+    OBJC_DISABLE_INITIALIZE_FORK_SAFETY
+    PYTHONPATH
```
Comment (felixxm) [inline tox.ini:28]: This change is unrelated, please revert.

File: tox.ini (hunk 2 above)
Comment (icemac) [reply tox.ini:18]: What a pity there is no ability to specify a minimum version for `basepython`. I my case it ran with Python3.7.
I am going to revert this change as the expectation seems to be that developers let `python3` point to a recent version.

File: tox.ini (hunk 3 above)
Comment (icemac) [reply tox.ini:28]: Okay, I'll create another PR with only this change then.

**Comment by icemac** [comment]
@felixxm I did the requested changes and also fixed the commit message. Please re-review.

**Review by felixxm — APPROVED** [review]
@icemac Thanks :+1: Welcome aboard :boat:
